lib/ansible/modules/windows/win_copy.ps1

```diff
@@ -76,7 +76,9 @@ Function Copy-File($source, $dest) {
         }
 
         if (Test-Path -Path $dest -PathType Leaf) {
-            Remove-Item -Path $dest -Force -Recurse | Out-Null
+            if (-not $check_mode) {
+                Remove-Item -Path $dest -Force -Recurse | Out-Null
```

Comment:
I would simply do:
```powershell
Remove-Item -Path $dest -Force -Recurse -WhatIf:$check_mode | Out-Null
```

Reply:
yup, got miscarried by earlier comments that avoided WhatIf because the path didn't exist; which is not the case here as the if conditional is Test-Path.... (facepalm)

Reply:
No worries. In reality there won't be a real difference, on the other hand if Microsoft at some point improves Remove-Item to also detect other failure-cases (in a safe way) the `-WhatIf` implementation may prove to be more useful.